api/v1beta1/types.go

```diff
@@ -123,6 +161,15 @@ type LoadBalancerSpec struct {
 	// +kubebuilder:validation:Pattern=`(^[1-9][0-9]{0,31}$)|(^[a-z][a-z0-9-]{4,28}[a-z0-9]$)`
 	// +optional
 	APIServerInstanceGroupTagOverride *string `json:"apiServerInstanceGroupTagOverride,omitempty"`
+
+	// LoadBalancerType defines the type of Load Balancer that should be created.
+	// If not set, a Global External Proxy Load Balancer will be created by default.
+	// +optional
+	LoadBalancerType *LoadBalancerType `json:"loadBalancerType,omitempty"`
```

Comment:
I initially thought that maybe this config lends itself well to being a discriminated union with `LoadBalancerType` being discriminated type. But, after reviewing the accompanying changes, I see some advantages to making this field []LoadBalancerType. 

LoadBalancerType(s) can then be populated with ExternalLoadBalancerOnly and/or InternalLoadBalancerOnly thus eliminating the need for `Dual`. This might simplify some of the checks in cloud/services/compute/loadbalancers/reconcile.go.